scripts/src/javatests/org/oppia/android/scripts/xml/XmlSyntaxErrorHandlerTest.kt

```diff
@@ -0,0 +1,88 @@
+package org.oppia.android.scripts.xml
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.xml.sax.SAXParseException
+import java.io.File
+import javax.xml.parsers.DocumentBuilder
+import javax.xml.parsers.DocumentBuilderFactory
+
+/** Tests for [XmlSyntaxErrorHandler]. */
+class XmlSyntaxErrorHandlerTest {
+  private val builderFactory = DocumentBuilderFactory.newInstance()
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("testfiles")
+  }
+
+  @Test
+  fun testXmlErrorHandler_validXml_noErrorShouldBeCollected() {
+    val validXml =
+      """
+      <?xml version="1.0" encoding="utf-8"?>
+      <shape xmlns:android="http://schemas.android.com/apk/res/android"
+        android:shape="rectangle">
+        <solid android:color="#3333334D" />
+        <size android:height="1dp" />
+      </shape>
+      """.trimIndent()
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(validXml)
+    val docBuilder = builderFactory.newDocumentBuilder()
+    val xmlSyntaxErrorHandler = XmlSyntaxErrorHandler()
+    docBuilder.setErrorHandler(xmlSyntaxErrorHandler)
+
+    parseXml(docBuilder = docBuilder, file = tempFile)
+
+    val errorList = xmlSyntaxErrorHandler.retrieveErrorList()
+    assertThat(errorList).isEmpty()
+  }
+
+  @Test
+  fun testXmlErrorHandler_invalidXml_errorListShouldBeEmpty() {
```

Comment:
Suggest adding tests for multiple errors, and for warning/error/fatal error cases (consider that if any lines of code in XmlSyntaxErrorHandler are commented out then at least 1 test should fail).

Reply:
Ah! I just dug a bit deep into this and realized that, for our use case, only `fatalError` is essential to us. Because, whenever there is any syntactically incorrect XML present, then always a `fatalError` is thrown, and for this case as we are only concerned of the syntax error hence, all our errors in testing will be of type `fatalError` only. 
Also, there won't be the case of multiple failures in a single XML file because, as soon as the parser encounters a `fatalError` it stops the further processing of XML.
Hence, it seems like we won't need tests for multiple failures as we will never get more than a single error and also the use cases of `warning` and `error` are irrelevant for our case. 
I was probably thinking of removing them from the syntax error handler but it is compulsory to override them.

Reply:
Added tests for all cases. PTAL.